rmw_test_fixture_implementation/src/rmw_test_fixture_default/rmw_test_fixture_default.c

```diff
@@ -182,7 +185,26 @@ rmw_test_isolation_start_default(void)
   // Avoid ROS_DOMAIN_ID=0 entirely
   slot += 1;
 
+  const char *old_env_val;
+  if (NULL == rcutils_get_env("ROS_DOMAIN_ID", &old_env_val)) {
+    fprintf(stderr, "Failed to get ROS_DOMAIN_ID\n");
+    port_lock_fini(g_lock);
+    g_lock = INVALID_PORT_LOCK;
+    return RMW_RET_ERROR;
+  }
+
   rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  if (NULL != g_restore_domain_id) {
+    allocator.deallocate(g_restore_domain_id, allocator.state);
```

Comment:
Should we add this to make sure the pointer is not deallocated twice?
```suggestion
    allocator.deallocate(g_restore_domain_id, allocator.state);
    g_restore_domain_id = NULL;
```

Reply:
Sure, but the next line will set it unconditionally anyway.